clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/task/datawriter/AbstractPartitionWriter.java

```diff
@@ -710,8 +595,33 @@ private String printDuplicateKey(byte[] keyBytes) {
 
     @Override
     public void close() {
-      Utils.closeQuietlyWithErrorLogged(recordReader);
-      Utils.closeQuietlyWithErrorLogged(keySerializer);
+      // Nothing to do
+    }
+  }
+
+  public class DelegatingProducerCallback implements PubSubProducerCallback {
```

Comment:
I suppose this is the same as the removed method ReducerProduceCallback?

Reply:
Yeah

Reply:
Actually, this was not true. The `ReducerProduceCallback` is still there in VeniceReducer. However, after I went through the implementation, I found I should move that callback and it's implementation into `AbstractPartitionWriter` and removed `DelegatingProducerCallback` completely.